apm-agent-plugins/apm-okhttp-test/src/test/java/co/elastic/apm/agent/okhttp/OkHttp3ClientVersionIT.java

```diff
@@ -0,0 +1,39 @@
+package co.elastic.apm.agent.okhttp;
+
+import co.elastic.apm.agent.TestClassWithDependencyRunner;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.util.Arrays;
+import java.util.List;
+
+@RunWith(Parameterized.class)
+public class OkHttp3ClientVersionIT {
+    private final TestClassWithDependencyRunner runner;
+
+    public OkHttp3ClientVersionIT(List<String> dependencies) throws Exception {
+        this.runner = new TestClassWithDependencyRunner(dependencies,
+            OkHttp3ClientInstrumentationTest.class);
+    }
+
+    @Parameterized.Parameters(name = "{0}")
+    public static Iterable<Object[]> data() {
+        return Arrays.asList(new Object[][]{
+            // ivy can't automatically resolve the dependencies, so we'll have to manually type them in
+            // check https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp for the respective versions of kotlin and okio
+            {List.of("com.squareup.okhttp3:okhttp:4.3.0", "org.jetbrains.kotlin:kotlin-stdlib:1.3.61", "com.squareup.okio:okio:2.4.1")},
```

Comment:
why not have a 1.x and 2.x versions in this list also ? I know that earlier version might not depend on Kotlin, thus we could expect them to be quite different code-wise.

Reply:
These are tested in `OkHttpClientInstrumentationTest.java` (not OkHttp3)